### PR TITLE
Interp sdb

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/mini-runtime.c [hunk 1]
```diff
@@ -4547,6 +4549,56 @@ mono_personality (void)
...
+	return MONO_BREAK_POLICY_ALWAYS;
+}
+
+static MonoBreakPolicyFunc break_policy_func = always_insert_breakpoint;
+
+/**
+ * mono_set_break_policy:
+ * \param policy_callback the new callback function
+ *
+ * Allow embedders to decide wherther to actually obey breakpoint instructions
```
Comment (iamthemovie) [inline mono/mini/mini-runtime.c]: Typo! wherther -> whether 😄

**Comment by lewurm** [comment]
build interpreter

File: mono/mini/interp/transform.c [hunk 2]
```diff
@@ -940,21 +1098,228 @@ interp_save_debug_info (RuntimeMethod *rtm, MonoMethodHeader *header, TransformD
...
+		printf ("\nSEQ POINT MAP FOR %s: \n", td->method->name);
+
+		for (i = 0; i < td->seq_points->len; ++i) {
+			SeqPoint *sp = (SeqPoint*)g_ptr_array_index (td->seq_points, i);
+			GSList *l;
+
+			if (!next [i])
+				continue;
+
+			printf ("\tIL0x%x[0x%0x] ->", sp->il_offset, sp->native_offset);
```
Comment (lewurm) [inline mono/mini/interp/transform.c]: please use `g_print`

**Comment by vargaz** [comment]
Note that some debugger tests are still failing. These include:
- domains
- invoke
- exception handling

**Comment by lewurm** [comment]
on linux CI fails due to that (that is without `--interpreter`):

```
* Assertion at interp/interp.c:227, condition `jit_tls' not met

Program received signal SIGABRT, Aborted.
0x00007ffff6efb428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
54      ../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  0x00007ffff6efb428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007ffff6efd02a in __GI_abort () at abort.c:89
#2  0x000000000083394a in mono_log_write_logfile (log_domain=0x0, level=G_LOG_LEVEL_ERROR, hdr=0,
    message=0xc11fe0 "* Assertion at interp/interp.c:227, condition `jit_tls' not met\n") at mono-log-common.c:135
#3  0x000000000082a81b in structured_log_adapter (log_domain=0x0, log_level=G_LOG_LEVEL_ERROR,
    message=0xc11fe0 "* Assertion at interp/interp.c:227, condition `jit_tls' not met\n", user_data=0x0) at mono-logger.c:432
#4  0x00000000008506da in monoeg_g_logv (log_domain=0x0, log_level=G_LOG_LEVEL_ERROR, format=0x87b378 "* Assertion at %s:%d, condition `%s' not met\n", args=0x7fffffffd760)
    at goutput.c:115
#5  0x000000000085088e in monoeg_assertion_message (format=0x87b378 "* Assertion at %s:%d, condition `%s' not met\n") at goutput.c:135
#6  0x0000000000566cff in set_context (context=0x0) at interp/interp.c:227
#7  0x000000000057aabb in mono_interp_init () at interp/interp.c:5274
#8  0x000000000041d541 in mini_init (
    filename=0x7fffffffdf88 "/home/lewurm/work/mono/external/roslyn-binaries/Microsoft.Net.Compilers/Microsoft.Net.Compilers.2.3.0/tools/csc.exe", runtime_version=0x0)
    at mini-runtime.c:3776
#9  0x00000000004c2751 in mono_main (argc=8, argv=0x7fffffffdca8) at driver.c:2077
#10 0x0000000000413b72 in mono_main_with_options (argc=8, argv=0x7fffffffdca8) at main.c:46
#11 0x0000000000414699 in main (argc=8, argv=0x7fffffffdca8) at main.c:339
```

not exactly sure why this isn't a problem on macOS.

**Pull request merged**